File: NHSOnline.Backend.Worker.Stubs/Journeys/Emis/Im1Connection/NoOnlineUserFoundJourney.cs
using System.Collections.Generic;
using NHSOnline.Backend.Worker.Mocking.Emis;
using NHSOnline.Backend.Worker.Mocking.Models;

namespace NHSOnline.Backend.Worker.Stubs.Journeys.Emis.Im1Connection
{
    internal class NoOnlineUserFoundJourney : Journey
    {
        internal override IEnumerable<Mapping> GetMappings()
        {
            const string odsCode = "A29928";
            const string surname = "Diaz";
            const string dateOfBirth = "1991-10-30T00:00:00";
            const string accountId = "3834473016";
            const string linkageKey = "QAkP7SUrfsQ1l";
            const string endUserSessionId = "7YjG1LYkOkSY1iAcXGG8ZU";

            yield return EndUserSessionConfigurator
                .ForRequest()
                .RespondWithSuccess(endUserSessionId);

            yield return MeConfigurator
                .ForRequest(endUserSessionId, surname, dateOfBirth, accountId, linkageKey, odsCode)
                .RespondWithNoOnlineUserFound();
        }
    }
}

File: bddtests/src/main/kotlin/mocking/defaults/dataPopulation/journies/im1Connection/NoOnlineUserFoundJourney.kt
package mocking.defaults.dataPopulation.journies.im1Connection

import mocking.MockingClient
import mocking.emis.me.LinkApplicationRequestModel
import mocking.emis.me.LinkageDetailsModel
import models.patients.EmisPatients

class NoOnlineUserFoundJourney(private val client: MockingClient) {
    val patient = EmisPatients.halleDawe

    fun create() {
        client
            .forEmis {
                authentication.endUserSessionRequest()
                        .respondWithSuccess(patient.endUserSessionId)
            }

        client
            .forEmis {
                authentication.meApplicationsRequest(
                        patient,
                        LinkApplicationRequestModel(
                                surname = patient.surname,
                                dateOfBirth = patient.dateOfBirth,
                                linkageDetails = LinkageDetailsModel(
                                        accountId = patient.accountId,
                                        linkageKey = patient.linkageKey,
                                        nationalPracticeCode = patient.odsCode
                                )
                        )
                ).respondWithNoOnlineUserFound()
            }
    }
}